plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -1115,15 +1116,36 @@ public void testDate(ZoneId sessionZone)
     public void testEnum()
     {
         JdbcSqlExecutor jdbcSqlExecutor = new JdbcSqlExecutor(postgreSqlServer.getJdbcUrl(), postgreSqlServer.getProperties());
-        jdbcSqlExecutor.execute("CREATE TYPE enum_t AS ENUM ('a','b','c')");
+        // Define enum values in a order different from lexicographical
+        jdbcSqlExecutor.execute("CREATE TYPE enum_t AS ENUM ('b', 'a', 'C')");
         jdbcSqlExecutor.execute("CREATE TABLE test_enum(id int, enum_column enum_t)");
         jdbcSqlExecutor.execute("INSERT INTO test_enum(id,enum_column) values (1,'a'::enum_t),(2,'b'::enum_t)");
         try {
             assertQuery(
                     "SELECT column_name, data_type FROM information_schema.columns WHERE table_schema = 'tpch' AND table_name = 'test_enum'",
                     "VALUES ('id','integer'),('enum_column','varchar')");
             assertQuery("SELECT * FROM test_enum", "VALUES (1,'a'),(2,'b')");
-            assertQuery("SELECT * FROM test_enum WHERE enum_column='a'", "VALUES (1,'a')");
+            assertThat(query("SELECT * FROM test_enum WHERE enum_column = 'a'"))
+                    .matches("VALUES (1, VARCHAR 'a')")
+                    .isFullyPushedDown();
+            assertThat(query("SELECT * FROM test_enum WHERE enum_column != 'a'"))
+                    .matches("VALUES (2, VARCHAR 'b')")
+                    .isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT * FROM test_enum WHERE enum_column <= 'a'"))
+                    .matches("VALUES (1, VARCHAR 'a')")
+                    .isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT * FROM test_enum WHERE enum_column <= 'b'"))
+                    .matches("VALUES (1, VARCHAR 'a'), (2, VARCHAR 'b')")
```

Comment:
Doesn't seem to match what I think Postgres would do. However since Trino doesn't have ENUM types and we map them to VARCHAR I think we are internally consistent and that's the best we can do for now.

Reply:
that's exactly why i added this test